js/src/Ice/WSTransceiver.js

```diff
@@ -6,333 +6,329 @@ import { ConnectionInfo as SSLConnectionInfo } from "./SSL/ConnectionInfo.js";
 import { SocketOperation } from "./SocketOperation.js";
 import { Timer } from "./Timer.js";
 
-export let WSTransceiver = {};
+//
+// With Chrome we don't want to close the socket while connection is in progress,
+// see comments on close implementation below.
+//
+// We need to check for Edge browser as it might include Chrome in its user agent.
+//
+const IsChrome = navigator.userAgent.indexOf("Edge/") === -1 && navigator.userAgent.indexOf("Chrome/") !== -1;
+const IsSafari = /^((?!chrome).)*safari/i.test(navigator.userAgent);
+
+const StateNeedConnect = 0;
+const StateConnectPending = 1;
+const StateConnected = 2;
+const StateClosePending = 3;
+const StateClosed = 4;
+
+class WSTransceiver {
+    constructor(instance, secure, addr, resource) {
+        this._readBuffers = [];
+        this._readPosition = 0;
+        this._maxSendPacketSize = instance.properties().getPropertyAsIntWithDefault("Ice.TCP.SndSize", 512 * 1024);
+        this._writeReadyTimeout = 0;
+
+        let url = secure ? "wss" : "ws";
+        url += "://" + addr.host;
+        if (addr.port !== 80) {
+            url += ":" + addr.port;
+        }
+        url += resource ? resource : "/";
+
+        this._url = url;
+        this._fd = null;
+        this._addr = addr;
+        this._desc = "local address = <not available>\nremote address = " + addr.host + ":" + addr.port;
+        this._state = StateNeedConnect;
+        this._secure = secure;
+        this._exception = null;
+    }
+
+    writeReadyTimeout() {
+        const t = Math.round(this._writeReadyTimeout);
+        this._writeReadyTimeout += this._writeReadyTimeout >= 5 ? 5 : 0.2;
+        return Math.min(t, 25);
+    }
+
+    setCallbacks(connectedCallback, bytesAvailableCallback, bytesWrittenCallback) {
+        this._connectedCallback = connectedCallback;
+        this._bytesAvailableCallback = bytesAvailableCallback;
+        this._bytesWrittenCallback = bytesWrittenCallback;
+    }
 
-if (typeof WebSocket !== "undefined") {
```

Comment:
Just removed this `if`, WebSocket is supported in modern Node.js, we don't support the older versions anymore.